movies/admin.py

```diff
@@ -32,20 +34,40 @@ def import_movies(self, request):
         form = CsvForm(request.POST, request.FILES)
         if form.is_valid():
             csv_file = TextIOWrapper(request.FILES['csv_file'], encoding=request.encoding)
-            movies = csv.reader(csv_file)
+            movies = csv.DictReader(csv_file)
+            new_movies = []
+            new_movies_pk = []
+            added_movies = []
             for movie in movies:
+                created_movie = Movie(**movie)
                 try:
-                    Movie.objects.create(name=movie[0],
-                                         poster=movie[1],
-                                         description=movie[2],
-                                         imdb_id=movie[3],
-                                         length_min=int(movie[4]),
-                                         trailer_url=movie[5])
+                    new_movies_pk.append(int(created_movie.pk))
+                    new_movies.append(created_movie)
+                    name = movie["poster"].split("/")[-1]
+                    response = requests.get(created_movie.poster)
+
+                    if response.status_code == 200:
+                        created_movie.poster.save(
+                            name, ContentFile(response.content), save=False
+                        )
 
                 except (requests.exceptions.MissingSchema, IntegrityError) as e:
                     messages.error(
                         request=request,
-                        message=f"\nFailed to add movie: {movie[0]} to the DB due to following error: {e}",
+                        message=f"\nFailed to add movie: {movie} to the DB due to following error: {e}",
                     )
 
+            system_list = Movie.objects.values_list('pk', flat=True)
+            for item in new_movies:
+                if int(item.pk) not in system_list:
+                    added_movies.append(item)
+            Movie.objects.bulk_create(added_movies)
+
+            if not Movie.objects.all():
+                Movie.objects.bulk_create(new_movies)
+            else:
+                Movie.objects.bulk_update(new_movies, fields=['name', 'poster', 'description', 'imdb_id', 'length_min', 'trailer_url'])
```

Comment:
Maybe I'm mistaken, but here it looks like you're also updating the added_movies objects for which you just called bulk_create.
Also, if you only need to check if there are any objects you can use the count method to use a lighter SQL query.